soh/src/code/z_sram.c

```diff
@@ -406,7 +406,7 @@ void Sram_InitSave(FileChooseContext* fileChooseCtx) {
                     gSaveContext.bgsFlag = true;
                 }
                 Item_Give(NULL, getItem.itemId);
-            } else if (getItem.modIndex == MOD_RANDOMIZER) {
+            } else if (getItem.modIndex == MOD_RANDOMIZER && getItem.getItemId != RG_ICE_TRAP) {
```

Comment:
I assume other instances calling Randomizer_Item_Give handle ice traps correctly?

Reply:
Yeah this normally doesn't get called at all when there's an ice trap. Some other piece of the control flow at that point always makes sure of that.

Reply:
Okay, this might be the issue with shopsanity as I don’t recall preventing it there